presto-main/src/main/java/com/facebook/presto/operator/unnest/ReplicatedBlockBuilder.java

```diff
@@ -30,48 +26,25 @@
 class ReplicatedBlockBuilder
 {
     private Block source;
-    private int[] ids;
-    private int positionCount;
+    private int sourcePosition;
 
     public void resetInputBlock(Block block)
     {
-        this.source = requireNonNull(block, "block is null");
+        source = requireNonNull(block, "block is null");
+        sourcePosition = 0;
```

Comment:
this assignment is redundant; 0 is the default value for int

Reply:
@mbasmanova This is necessary because we need to reset sourcePosition to 0 every time a new page comes in.